src/sudoers/tokens.rs

```diff
@@ -194,7 +194,12 @@ impl Token for EnvVar {
     }
 
     fn accept(c: char) -> bool {
-        c.is_ascii_alphanumeric() || "*_%".contains(c)
+        !c.is_control() && !c.is_whitespace() && !Self::escaped(c)
+    }
+
+    const ESCAPE: char = '\\';
+    fn escaped(c: char) -> bool {
+        "\\=#\"".contains(c)
```

Comment:
nit: this is a compact way to write this but I found it a bit hard to parse out the four characters in the string because of the escaping. I think `matches!(c, '\\' | '=' | '#' | '"')` would be easier to parse, even if it's longer to type